odd_collector/adapters/druid/client.py

```diff
@@ -0,0 +1,125 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from aiohttp import ClientSession
+from odd_collector_sdk.errors import DataSourceError
+
+from odd_collector.adapters.druid.domain.column import Column
+from odd_collector.adapters.druid.domain.table import Table
+from odd_collector.domain.plugin import DruidPlugin
+
+
+class DruidBaseClient(ABC):
+    @abstractmethod
+    async def get_tables(self) -> List[Table]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_columns(self) -> List[Column]:
+        raise NotImplementedError
+
+    async def get_tables_nr_of_rows(self) -> dict:
+        raise NotImplementedError
+
+
+class DruidClient(DruidBaseClient):
+    def __init__(self, config: DruidPlugin) -> None:
+        self.__config = config
+
+    async def get_tables(self) -> List[Table]:
+        """
+        Get tables from druid.
+
+        Note: It excludes the INFORMATION_SCHEMA that keep druid metadata related
+        to tables persisted.
+
+        Note: It excludes system tables related to druid operations.
+
+        :return: a list of tables
+        """
+        # Prepare
+        url = f"{self.__config.host}:{self.__config.port}/druid/v2/sql/"
+        sql_query = """
+            SELECT 
+              "TABLE_CATALOG", 
+              "TABLE_SCHEMA", 
+              "TABLE_NAME", 
+              "TABLE_TYPE", 
+              "IS_JOINABLE", 
+              "IS_BROADCAST"
+            FROM INFORMATION_SCHEMA.TABLES
+            WHERE LOWER("TABLE_SCHEMA") != 'information_schema'
+              AND LOWER("TABLE_SCHEMA") != 'sys'
+        """
+
+        # Execute
+        try:
+            async with ClientSession() as session:
```

Comment:
One more moment :)
Methods: `get_tables`, `get_columns` and `get_tables_nr_of_rows` each time create new session, it will be better to create session once.

As a possible solution:
1. Add session  parameter `session: ClientSession` to methods
2. Create method which will manage connection and call them, i.e
```python
# client.py
async def get_resources(self):
    async with ClientSession() as session:
        tasks = [
            asyncio.create_task(self.get_tables(session)),
            asyncio.create_task(self.get_columns(session)),
            asyncio.create_task(self.get_tables_nr_of_rows(session))
        ]

        return asyncio.gather(*tasks)

...
# adapter.py
class Adapter:
    ...
    async def get_data_entities(self) -> List[DataEntity]:
        # Fetch
        ...
        tables, columns, tables_nr_of_rows = await self.client.get_resources()
        ...
  
```

Reply:
Nice 👍, will do